### PR TITLE
Support Logging for vmcall-raw feature

### PR DESCRIPTION
**Pull request by mgudaram**

### CODE REVIEW

File: src/migtd/src/migration/session.rs [hunk 1]
```diff
@@ -46,6 +54,21 @@ const GSM_FIELD_MAX_EXPORT_VERSION: u64 = 0x2000000100000002;
...
+
+#[cfg(feature = "vmcall-raw")]
+static LOGAREA_INITIALIZED: AtomicBool = AtomicBool::new(false);
+
+#[cfg(feature = "vmcall-raw")]
+static LOGENTRY_ID: AtomicU64 = AtomicU64::new(0);
+
+#[cfg(feature = "vmcall-raw")]
+static MAXLOGLEVEL: AtomicU8 = AtomicU8::new(0);
+
```
Comment (jyao1) [inline src/migtd/src/migration/session.rs]: can we create a data structure to include all logging related items?

**Comment by jyao1** [comment]
Please clarify how you do the unit test. If you have covered all corner cases, including
* log first
* log with enough space in the end
* log without enough space in the end and with enough space from the head
* log without enough space in the end and even without enough space from the head
* ...

File: src/migtd/src/migration/session.rs [hunk 2]
```diff
@@ -430,24 +534,218 @@ pub async fn get_tdreport(
...
+            Level::Debug,
+            request_id,
+        );
         return Err(MigrationResult::InvalidParameter);
     }
     Ok(())
 }
 
+#[cfg(feature = "vmcall-raw")]
+pub fn create_logarea() -> Result<()> {
```
[the remark this answers is not in the document]
Comment (mgudaram) [reply src/migtd/src/migration/session.rs]: > can we move all logging related feature to a separate log.rs file?

sure jiewen, since session.rs uses existing crate log::, is it okay to rename the file to **logging.rs**? it would help resolve any future confusions.

File: src/migtd/src/migration/session.rs (hunk 1 above)
Comment (mgudaram) [reply src/migtd/src/migration/session.rs]: > can we create a data structure to include all logging related items?

I moved them into a data structure in logging.rs

File: src/migtd/src/migration/session.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (mgudaram) [reply src/migtd/src/migration/session.rs]: sounds good. I have updated the code. Thanks.

**Comment by mgudaram** [comment]
> > Please clarify how you do the unit test. If you have covered all corner cases, including
> > 
> > * log first
> > * log with enough space in the end
> > * log without enough space in the end and with enough space from the head
> > * log without enough space in the end and even without enough space from the head
> > * ...
> 
> On my end, I have run ~50 migrations with logging using loopback mechanism. Few migrations exhaust the circular buffer. I will look into adding unit test support in the logging.rs.

I have updated the unit tests as mentioned. Thanks!

**Comment by jyao1** [comment]
please resolve conflict

**Review by jyao1 — CHANGES_REQUESTED** [review]
please dont use merge.
